src/targets/gpu/compile_gen.cpp

```diff
@@ -75,25 +82,26 @@ std::string vectorize::str() const
 preload preload::broadcasts(std::size_t axis, const std::vector<shape>& inputs)
 {
     const std::size_t max_lds_bytes = 4096;
-    std::vector<bool> result;
-    std::transform(inputs.begin(),
-                   inputs.end(),
-                   std::back_inserter(result),
-                   [&](const shape& input) { return input.strides()[axis] == 0; });
-    auto bytes = std::inner_product(inputs.begin(),
-                                    inputs.end(),
-                                    result.begin(),
-                                    std::size_t{0},
-                                    std::plus<>{},
-                                    [](const shape& s, bool b) -> std::size_t {
-                                        if(b)
-                                            return s.bytes();
-                                        return 0;
-                                    });
-    if(bytes < max_lds_bytes)
-        return {result};
-    // TODO: Try to partially preload items
-    std::fill(result.begin(), result.end(), false);
+    std::vector<bool> result(inputs.size());
+    std::vector<std::size_t> preloaded;
+    for(auto i : range(inputs.size()))
+    {
+        if(inputs[i].strides()[axis] == 0)
+            preloaded.push_back(i);
+    }
```

Comment:
If we want to avoid the cppcheck failure, this could be changed to: 
```
auto idxs = range(inputs.size());
std::copy_if(idxs.begin(), idxs.end(), std::back_inserter(preloaded), [&](auto i){ return inputs[i].strides()[axis] == 0; });
```

@kahmed10 What are your thoughts? Is that less readable?

Reply:
I thought it could be rewritten as:
```
std::copy_if(inputs.begin(), inputs.end(), std::back_inserter(preloaded), [&](auto input){ return input.strides()[axis] == 0; });
```